src/v/cluster/controller.json

```diff
@@ -135,6 +135,11 @@
             "name": "get_partition_state",
             "input_type": "partition_state_request",
             "output_type": "partition_state_reply"
+        },
+        {
+            "name": "get_controller_last_applied",
```

Comment:
Isn't that the idea though? We want to return the last applied and wait on that. If the stm is lagging, then thats the same response that would have been returned if we had invoked the command on the leader anyway.

Reply:
not sure I follow.. the specific case I had in my mind is change of leadership, if this request hits a leader different than the one that actually committed+applied the ACL request, there is no guarantee that the last_applied >= committed_offset of acl batch. Consider the following sequence of actions.

- create/delete-acl - broker 0 (current controller_leader) - raft committed_offset = applied_offset = 10 (command waits until applied_offset = committed_offset so that is guaranteed)
- leadership change - broker 1 (new controller leader) - raft_committed_offset is at least 10 (raft leadership guarantee) but there is no guarantee on applied_offset, theoretically it could be lagging
- now a follower makes a request to broker 1, it returns a stale applied_offset

Reply:
I see your point, so if at any given time an elected leader applied the command, the effect is considered as performed but in the situation you described we could still be reporting stale state +1